### ISSUE TITLE
admin panel totally inaccessible for screen readers

### ISSUE THREAD

**Issue by ghost** [magento/magento2#23931]
i'm thadeu, a blind ecommerce owner. and i was wondering to get started with magento... but the problem is:

the magento admin panel, has mouseover elements, or something similar to it...
what happens is: i can't do anything using a screen reader. since i'm blind, i can't do anything.

i'm using latest version of magento...
what this was suposed to happen is, soon as i put the screen reader cursor in the catalog, it should open the next options in a list or something to it, my jaws screen reader said that this element contains a list, so, it should open with the mouse cursor inside of it, but, it does not.
i guess the menus should work as wordpress does, so, if i do use the keuyboard, the lists popups automatically, without any requiered click...
i don't know why do this is happening to me...
another thing, i can't save settings to with a screen reader, i guess magento uses a lot of unnecessary javascript. and javascript used incorrectly, breaks the full accessibility...
in the magento docs, there are a lot of things about accessibility for sellers, but, this is stupid, but, it is not accessible in the admin panel...
and i did also used NVDA, same issue...
how to solve: do not use mouseover elements, or, make the menus apears once clicked with the mouse, and alson if we hit enter on it.
i'm glad to send any programming report if required...

note: fix this more soon as possible... it happens in magento open source, i don't know about magento commerce, because i don't use this software...

**Comment by engcom-Charlie** [comment]
Hi @thadeu220 ! What software do you use?

**Comment by ghost** [comment]
I do use NVDA, a free screen reader with all popular browsers such as Mozilla firefox, google chrome and also internet explorer
I did not tested on mobile browsers with their specific operating system such safari
Since i am new to github, i don't know how to put labels on this topic lol :D
Well i did the same tests with jaws but same issue.
NBDA and jaws are in the same browsers and i did the test in another computers.nvda are free, you may download it @
Www.nvaccess.org
It also offers debbuging and speetch viewer.
You dont need to install it, cause it runs in portable versions.
Jaws requires a licensed version but it could be crack

Note: the issue are in the Admin nav menus.

**Comment by engcom-Charlie** [comment]
@thadeu220 , sorry, but i can't install NVDA because my OS is Linux.  
So I can only advise using another software(Bandicam is good).
I have to close this issue.
Thank for your report!

**Comment by ghost** [comment]
i will reopen it... because this in no sense... (Bandicam is a screen recorder, and we are talking about screen readers...
and linux has orca, you may install it running this in the terminal, consider in this sample, the ubuntu...
sudo apt install orca
than
sudo orca

this will install and activate orca screen reader... the problem are in screen reader, due to accessibility problems.
the mobile browsers situation are also confirmed, i do used also mac OSX and same issue.
PS: i'm using hostinger as a hosting company, and i installed the magento environment using the terminal, uploaded files, and unzip with terminal.
after unzip, i did the installation using  my browser, the installation workflow works perfectly.
so, the problems are in the menu section.
for mac osX we hacve voiceover.
for linux, we do have orca screen reader
and windows are NVDA, dolphin, jaws and virtual vision (brazillian screen reader, it crashes and get lags more than helping blind computer's users).
please, since this is not off-topic, and it is a serious behavior, do not close this issue while it is not fixed...
i will be helping all members envolved.
thanks.

PS: I'm still suspecting, but, where can i host, for free a magento website, for testing? i do have a vps and a reseller panel, but, there's no way to host there...
any free hosting or something that will help me?

**Comment by ghost** [comment]
i'm looking for another web hosting, just to test if the problem are in my hosting (this is pratically impossible, but...) or, in the magento...
cause, i really don't know, i do tested the magento live demo, and it works, that's why i'm asking.

**Comment by engcom-Charlie** [comment]
@thadeu220 i tested with ChromeVox extension https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en
and this extension works fine on admin page. I can propose to use it with Magento.

**Comment by ghost** [comment]
friend, i did the test, and it reads...
nvda also reads the text normally. but you cannot use the menus in the navigation bar, even with chrome vox... and chromevox does the same thing.
the issue is: you cannot navigate the navigation menus with the keyboard...
if i move the nvda focus to the catalog, instead of popping up the add product and other function, it does nothing...
neighter opening the lists...
the problem are not i'm my browser, and chrome vox, same situation...
it reads, but  i can't navigate through the options...
can someone install a clean maggento for me, and give me the credentials through my personal e-mail? so i can test...
cause, i don't know, sometimes i do thing the problem are in my web hosting...
but, it does not make any sense...
well... if someone does me a clean install and give me the credentials for testing, it will be great, so i will be 100% sure about the problem, if it is in the magento, or my hosting, or computer.
thanks...
and again, we are in the testing side...
if someone knows a free hosting where can i put my magento, for testing, tell me;;;
thanks in advance.

**Comment by engcom-Charlie** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-Charlie. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @engcom-Charlie, here is your [Magento instance](https://i-23931-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23931-2-3-develop.instances.magento-community.engineering/admin](https://i-23931-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ghost** [comment]
THANKS... LETS SEE

**Comment by engcom-Charlie** [comment]
@thadeu220 i understand your problem and i find it on admin pages.

**Comment by engcom-Charlie** [comment]
@thadeu220 i reopen this issue

**Comment by ghost** [comment]
hwhat the fuck! it works...
confirmed, starring, issue finished, the problem are in my environment, but, the problem, i don't know witch one...
i'm using hostinger... PHP 7.1

**Comment by engcom-Charlie** [comment]
@thadeu220 did you solve the problem?

**Comment by ghost** [comment]
in the development instance that you gave me, it works normally, but, it does not work in my hosting...

**Comment by engcom-Charlie** [comment]
@thadeu220 so i have to close it. 
Thank you for your report and contributing!

**Comment by antonkril** [comment]
Hi @thadeu220,

Have you found the cause of the problem? We can still help you troubleshoot it.

**Comment by ghost** [comment]
the problem was found... unfortunatelly the problem was my hosting... i have no idea what was going on...
but in another server, it works...
but i guess it is really complex to manage magento... i guess it is supossed to have an easier panel, just like wordpress does...
in wordpress / woocommerce, the areas are separated with mouseover elements, followed by a list... and there is only necessary things...
i don't think the multistore should be enabled by default...
and pratically everything i do i must clear the freaking cache all the time, and this is really time consuming...
my new hosting has no SSH access... how to setup the crontab using cpanel / directadmin?
in the cron job window, it asks the command, i have no idea what to insert...
running the latest version....
is there a way to autoclear the cache?
and last thing... how to change the languages?

issue fixed...
now, these are my questions... cause documentation are not so friendly with shared no ssh hostings.

**Comment by ghost** [comment]
aaaaaaaah... i do discovered something...
if magento works, my screen reader says:
"clickable configuration men linku"
if it are not working, it says:
"configuration menu link"

so... the elements are in the clickable / onclick status... if this is not happening, it means the screen reader will not be able to clcik
i also asked for a sighted person to try, and no clicking,... so... problem regarding cache or something like this...
i do think the cache must be disabled by default....
so crappy

**Comment by antonkril** [comment]
Have you disabled Page and Block caches on your instance?

**Comment by ghost** [comment]
How can I disable the cache if i cant neighter modify any setting.
I do discovered the cache because i flush cache using SSH in a vm box
Nut some hosting does not offer SSH and no terminal
Only cpanel

**Comment by ghost** [comment]
And magento should be more lulight-weight and easier to use
The docs reffers omly SSH, but not enough instructions for non-SSH companies

**Comment by Rus0** [comment]
You'll need to System > Cache Management

But since your menu is not working, I suggest contacting your hosting provider and ask them to do via shell command, and try again. Anything else, please free to message me, I will assist you.

**Comment by ghost** [comment]
@Rus0 i will try again...
since you're good enough to assist me, and i am in the really beggining in magento, i will give you my whatsapp number, if you want to help me, i prefer via whatsapp...
you can send me voice messages and also texts...
here it goes:
+55 35 992164289

i really need help to get started, and someone to explain me the basis, it will be really apreciatefull.

thanks.
